### PR TITLE
Add a web framework for the Rust language

### PR DESCRIPTION
**Pull request by Jzow**
CC [issues 72](https://github.com/wansenai/summer-boot/issues/72).

### CODE REVIEW

**Review by waghanza — CHANGES_REQUESTED** [review]
Maybe you miss some configuration in config.yaml, some yaml files must be uploaded to the container.

You can take inspiration from spring config file for static_files configuration

**Comment by Jzow** [comment]
[spring config.yaml](https://github.com/the-benchmarker/web-frameworks/blob/master/java/spring/config.yaml) From a language perspective, they are different, and we currently need to do a lot of work here while scanning the configuration file

**Comment by waghanza** [comment]
Sorry, you need to add
```yaml
files:
  - resources/**/*.yml
```
in config.yaml in order to load file onto container at compilation stage

**Comment by Jzow** [comment]
> Sorry, you need to add
> 
> ```yaml
> files:
>   - resources/**/*.yml
> ```
> 
> in config.yaml in order to load file onto container at compilation stage

Understood, thanks

File: rust/summer-boot/config.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+framework:
+  github: wansenai/summer-boot
+  version: 1.4
+  files:
+    - resources/**/*.yml
```
Comment (waghanza) [inline rust/summer-boot/config.yaml:5]:
```suggestion
    - src/resources/**/*.yml
```

**Review by waghanza — CHANGES_REQUESTED** [review]

File: rust/summer-boot/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+[package]
+name = "summer-boot"
```
Comment (waghanza) [inline rust/summer-boot/Cargo.toml:2]:
```suggestion
name = "server"
```

**Review by waghanza — COMMENTED** [review]
Could you unify compilation options for rust frameworks ?

File: rust/summer-boot/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+use summer_boot::{Request, Result};
+
+#[summer_boot::auto_scan()]
+#[summer_boot::main]
+async fn main() {
+    summer_boot::run();
+}
+
+#[summer_boot::get("/hello")]
```
Comment (waghanza) [inline rust/summer-boot/src/main.rs:9]:
```suggestion
#[summer_boot::get("/")]
#[summer_boot::post("/user")]
```

File: rust/summer-boot/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+pub async fn hello(mut _req: Request<()>) -> Result {
+    Ok(format!("Hello, Summer Boot").into())
```
Comment (waghanza) [inline rust/summer-boot/src/main.rs:11]:
```suggestion
    Ok().into())
```

**Review by waghanza — COMMENTED** [review]
Missing to implement GET on /user/:I'd that returns :id

**Comment by Jzow** [comment]
Okay, I'll make the changes later

**Pull request merged**